routers/api/v1/repo/repo.go

```diff
@@ -746,6 +748,15 @@ func updateBasicProperties(ctx *context.APIContext, opts api.EditRepoOption) err
 		return err
 	}
 
+	if updateRepoLicense {
+		if err := repo_module.AddRepoToLicenseUpdaterQueue(&repo_module.LicenseUpdaterOptions{
```

Comment:
Move them info `UpdateRepository` or we should have a new function named `SetRepositoryDefaultBranch`.

Reply:
I think we already have one. API and web UI uses different approachs, but they can be merged.

Reply:
We don't know whether default branch has been changed, so it can not be moved into `UpdateRepository`
We already have a function `repo_services.SetRepoDefaultBranch`, which will update DB record, git repo, and licenses.
But it will cause multiple DB update, as we will call `UpdateRepository` below.
So is it necessary to add a new function for this special usage case?